src/screens/Asset/ReceiveModal.js

```diff
@@ -119,34 +132,38 @@ export default class ReceiveModal extends React.Component<Props, *> {
             <View
               style={{
                 overflow: 'hidden',
+                padding: 10,
               }}
             >
               <QRCodeWithTheme value={address} size={160} />
             </View>
           </QRCodeWrapper>
-          <Button
-            title="Share Address"
-            onPress={this.handleAddressShare}
-            style={{
-              marginBottom: 20,
-              marginTop: spacing.mediumLarge,
-            }}
-          />
-        </ContentWrapper>
-        {showBuyTokensSection && (
-          <BuyTokensWrapper>
+          <ButtonsRow>
+            {showBuyTokensButton && (
+              <Button
+                title="Buy tokens"
+                onPress={handleBuyTokens}
+                positive
+                width={buttonWidth}
+                small={needsSmallButtons}
+              />
+            )}
             <Button
-              title="Buy tokens"
-              onPress={handleBuyTokens}
-              positive
+              title="Share Address"
+              onPress={this.handleAddressShare}
+              width={buttonWidth}
+              small={needsSmallButtons}
+              block={!buttonWidth}
             />
-            <IconsContainer>
-              <Image source={visaIcon} />
-              <IconsSpacing />
-              <Image source={mastercardIcon} />
-            </IconsContainer>
-          </BuyTokensWrapper>
-        )}
+          </ButtonsRow>
+          {showBuyTokensButton &&
```

Comment:
```suggestion
          {showBuyTokensButton && (
```
to keep consistency with the changes above :)